tests/integration_tests/functional/test_api.py

```diff
@@ -940,52 +945,76 @@ def test_api_version(test_microvm_with_api):
     assert out.strip()[1:] == preboot_response.json()['firecracker_version']
 
 
-def test_api_vsock(test_microvm_with_api):
+def test_api_vsock(bin_cloner_path):
     """
     Test vsock related API commands.
 
     @type: functional
     """
-    test_microvm = test_microvm_with_api
-    test_microvm.spawn()
-    test_microvm.basic_config()
+    builder = MicrovmBuilder(bin_cloner_path)
+    artifacts = ArtifactCollection(_test_images_s3_bucket())
+    # Fetch 0.25 and older firecracker binaries.
+    # Create a vsock device with each FC binary
+    # artifact.
+    firecracker_artifacts = artifacts.firecrackers(
+        # v1.0.0 deprecated `vsock_id`.
+        min_version="1.0.0")
+
+    for firecracker in firecracker_artifacts:
+        firecracker.download()
+        jailer = firecracker.jailer()
+        jailer.download()
+
+        vm_instance = builder.build_vm_nano(
+                                        fc_binary=firecracker.local_path(),
+                                        jailer_binary=jailer.local_path())
+
+        _test_vsock(vm_instance.vm)
+
+
+def _test_vsock(vm):
+    # Create a vsock device.
+    response = vm.vsock.put(
+        guest_cid=15,
+        uds_path='vsock.sock'
+    )
+    assert vm.api_session.is_status_no_content(response.status_code)
 
-    response = test_microvm.vsock.put(
+    # Updating an existing vsock is currently fine.
+    response = vm.vsock.put(
+        guest_cid=166,
+        uds_path='vsock.sock'
+    )
+    assert vm.api_session.is_status_no_content(response.status_code)
+
+    # Check PUT request. Although vsock_id is deprecated, it must still work.
+    response = vm.vsock.put(
         vsock_id='vsock1',
         guest_cid=15,
         uds_path='vsock.sock'
     )
-    assert test_microvm.api_session.is_status_no_content(response.status_code)
+    assert vm.api_session.is_status_no_content(response.status_code)
+    assert response.headers['deprecation']
```

Comment:
Having this check we only know that there are deprected fields, buy we don't know which ones. Should we also check the response message ?

Reply:
There is no message in the response, it's only in the logs. Do you still think we should check for it? I don't find it particularly useful.

Reply:
I see. Then I think it's ok to leave it as it is.